ext/standard/string.c

```diff
@@ -1813,8 +1821,8 @@ PHP_FUNCTION(stristr)
 	ZEND_PARSE_PARAMETERS_END();
 
 	if (!ZSTR_LEN(needle)) {
-		php_error_docref(NULL, E_WARNING, "Empty needle");
-		RETURN_FALSE;
+		zend_throw_exception(zend_ce_error_exception, "Empty needle", E_ERROR);
+		return;
```

Comment:
I think we might be better off removing these errors entirely instead. There is nothing semantically wrong with an empty needle (it'll always match at the first position). It would remove an unnecessary special case.

Reply:
Working on it the various implementation of str*pos function family make it a bit awkward to do it in this PR (maybe?). So I will probably do this change separately and try to clean up a bit the implementations.

Reply:
Yeah, it would be best to do this in a separate PR.